File: src/main/java/com/example/ssauc/admin/entity/Reply.java
package com.example.ssauc.admin.entity;

import com.example.ssauc.user.contact.entity.Board;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@ToString
@Table(name = "reply")
public class Reply {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "reply_id")
    private Long replyId;

    @OneToOne
    @JoinColumn(name = "board_id", nullable = false)
    private Board board;

    @ManyToOne
    @JoinColumn(name = "replier_id", nullable = false)
    private Admin admin;

    @Column(name = "subject", nullable = false, length = 200)
    private String subject;

    @Column(name = "message", columnDefinition = "TEXT")
    private String message;

    @Column(name = "complete_at")
    private LocalDateTime completeAt;
}

File: src/main/java/com/example/ssauc/admin/entity/Reply.kt
package com.example.ssauc.admin.entity

import com.example.ssauc.user.contact.entity.Board
import jakarta.persistence.*
import lombok.*
import java.time.LocalDateTime

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@ToString
@Table(name = "reply")
class Reply {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "reply_id")
    public var replyId: Long? = null

    @JvmField
    @OneToOne
    @JoinColumn(name = "board_id", nullable = false)
    public val board: Board? = null

    @ManyToOne
    @JoinColumn(name = "replier_id", nullable = false)
    public val admin: Admin? = null

    @Column(name = "subject", nullable = false, length = 200)
    public var subject: String? = null

    @Column(name = "message", columnDefinition = "TEXT")
    public var message: String? = null

    @Column(name = "complete_at")
    public var completeAt: LocalDateTime? = null
}
